### PR TITLE
HDDS-1659. Define the process to add proposal/design docs to the Ozone subproject

### PR DESCRIPTION
**Pull request by elek**
We think that it would be more effective to collect all the design docs in one place and make it easier to review them by the community.

We propose to follow an approach where the proposals are committed to the hadoop-hdds/docs project and the review can be the same as a review of a PR

See: https://issues.apache.org/jira/browse/HDDS-1659

### CODE REVIEW

**Comment by elek** [comment]
The first version (#922) is reverted due to a RAT failure. But I also updated the text itself to address some earlier comments.

 1. typo is fixed (thanks @ajayydv)
 2. I clearly defined that the design docs which are not yet implemented shouldn't be visible on the docs page (or only on a separated page) (thanks @eyanghwx)
 3. I clarified that we would like to be compatible with the existing practices (creating PDF with any method or create a wiki page should be accaptable) but I suggested to create a reference page with a summary and link. I think it addresses the possible formatting issues (thanks @arp7, what do you think about this approach?)

TL;DR:

 * Create a new jira + a  new page under the `docs/content/design` folder for each new design docs
 * Either include the content of the design in the markdown file (recommended as it can be commented line by line) or include the link (to Apache Jira attachment or Apache Wiki)

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 31 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 508 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1312 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 463 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 681 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 61 | The patch does not generate ASF License warnings. |
| | | 2749 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite yamllint |
| uname | Linux d1226530be24 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 23c0379 |
| Max. process+thread count | 415 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 35 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 488 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1260 | branch has no errors when building and testing our client artifacts. |
| -0 | patch | 1342 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 446 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 673 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 56 | The patch does not generate ASF License warnings. |
| | | 2636 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite |
| uname | Linux d1de1989d91f 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4e66cb9 |
| Max. process+thread count | 447 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 99 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 610 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1488 | branch has no errors when building and testing our client artifacts. |
| -0 | patch | 1554 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 593 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 756 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 47 | The patch does not generate ASF License warnings. |
| | | 3145 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite |
| uname | Linux 9034a57873cb 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c0a0c35 |
| Max. process+thread count | 310 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 40 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 629 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1409 | branch has no errors when building and testing our client artifacts. |
| -0 | patch | 1469 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 586 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 676 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 2903 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite |
| uname | Linux 28995c9521a1 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e20b195 |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/4/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 599 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1425 | branch has no errors when building and testing our client artifacts. |
| -0 | patch | 1530 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 572 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 633 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 2905 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite |
| uname | Linux 389772af3438 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 70b4617 |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/5/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 44 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 632 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1406 | branch has no errors when building and testing our client artifacts. |
| -0 | patch | 1477 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 563 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 603 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 2813 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-950/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/950 |
| Optional Tests | dupname asflicense mvnsite |
| uname | Linux 2ece94e3d99a 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 397a563 |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-950/6/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.
